### PR TITLE
Add Encrypted Transport resources

### PR DESCRIPTION
**Pull request by jrspinella**
## Describe your changes

 1. Added new route tabler and route for AFW subnet to be used with encrypted transport add on.
 2. Added variables to support the route 
 3. Fixed NW not to create but to lookup existing NW
4. Removed the 'create_network_watcher' 
5. Updated examples to have a commerical and government folders. 
6. Removing not used varaibles
7. Rename Scaffolding file

## Issue number

#29 #18 #19 #34

## Testing
This PR was tested on Commerical and US Gov clouds

## Checklist before requesting a review
- [x] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [x] I have executed pre-commit on my machine
- [x] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

**Comment by sstjean** [comment]
Created a new Example to test this.  `examples/hub_w_force_tunnel_and_encrypted_transport` which is a copy of `examples/hub_w_force_tunnel'`.  Getting an error when running a `terraform init`.

```
There are some problems with the configuration, described below.

The Terraform configuration must be valid before initialization so that
Terraform can determine which modules and providers need to be installed.
╷
│ Error: Module is incompatible with count, for_each, and depends_on
│
│   on main.tf line 56, in module "mod_vnet_hub":
│   56:   depends_on = [module.mod_logging]
│
│ The module at module.mod_vnet_hub is a legacy module which contains its own local provider
│ configurations, and so calls to it may not use the count, for_each, or depends_on
│ arguments.
│
│ If you also control the module "../..", consider updating this module to instead expect
│ provider configurations to be passed by its caller.
╵
```

**Comment by jrspinella** [comment]
> Created a new Example to test this. `examples/hub_w_force_tunnel_and_encrypted_transport` which is a copy of `examples/hub_w_force_tunnel'`. Getting an error when running a `terraform init`.
> 
> ```
> There are some problems with the configuration, described below.
> 
> The Terraform configuration must be valid before initialization so that
> Terraform can determine which modules and providers need to be installed.
> ╷
> │ Error: Module is incompatible with count, for_each, and depends_on
> │
> │   on main.tf line 56, in module "mod_vnet_hub":
> │   56:   depends_on = [module.mod_logging]
> │
> │ The module at module.mod_vnet_hub is a legacy module which contains its own local provider
> │ configurations, and so calls to it may not use the count, for_each, or depends_on
> │ arguments.
> │
> │ If you also control the module "../..", consider updating this module to instead expect
> │ provider configurations to be passed by its caller.
> ╵
> ```

I will take a look

File: README.md [hunk 1]
```diff
@@ -556,40 +577,41 @@ By default, this module will create a resource group and the name of the resourc
...
-* Azure Key Vault
-* Azure Application Gateway
-* Azure Load Balancer
-* Azure Network Security Group
-* Azure Virtual Network Gateway
-* Azure Virtual Network
+## Enable Force Tunneling for the Firewall
 
-> **NOTE:**  Please review the [Mission Enclave Policy Starter](https://github.com/azurenoops/ref-scca-enclave-policy-starter) reference implementation for more information.
+By default, this module will not create a force tunnel on the firewall. You can enable/disable it by appending an argument `enable_force_tunneling` located in `variables.fw.tf` If you want to enable a DDoS plan using this module, set argument `enable_force_tunneling = true`. Enabling this feature will ensure that the firewall is the default route for all the T0 through T3 Network routes.
```
Comment (sstjean) [inline README.md:592]: The third sentence says "If you want to enable a DDoS Plan..."

Comment (jrspinella) [reply README.md:592]: fixed

File: versions.tf [hunk 2]
```diff
@@ -22,5 +22,6 @@ provider "azurerm" {
   alias           = "ops_network"
   subscription_id = coalesce(var.ops_subscription_id, data.azurerm_client_config.current.subscription_id)
   environment     = var.environment
+  skip_provider_registration = var.environment == "usgovernment" ? true : false # Azure Government Cloud does not support some of the providers used in this module
```
Comment (sstjean) [inline versions.tf:25]: Should this comment reflect that all the providers we need are available in Azure Gov?

**Review by matross-gh — APPROVED** [review]
Approving on the basis of the change needed to unblock testing.  For future PRs, please limit the number file changes to 1 or 2 and increase the number of commits on your branch to help the PR review process.

**Review by matross-gh — APPROVED** [review]

**Review by sstjean — APPROVED** [review]
Reviewed code with @jrspinella and tested the `/examples/Commercial/managment_hub_w_force_tunnel_ddos_encrypted_transport` example.  Code looks good and `init`, `validate`, `plan`, and `destroy` all worked as expected.

**Pull request merged**